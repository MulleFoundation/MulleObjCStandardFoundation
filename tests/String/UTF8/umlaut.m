//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
//#import "MulleStandaloneObjCFoundation.h"



static mulle_utf8_t    hoehoe[] =
{
   0x22, 0x48, 0xc3, 0xb6, 0x68, 0xc3, 0xb6, 0x68,
   0xc3, 0xb6, 0x2c, 0x20, 0x77, 0x69, 0x72, 0x20,
   0x73, 0x69, 0x6e, 0x64, 0x20, 0x64, 0x69, 0x65,
   0x20, 0x47, 0x72, 0xc3, 0xb6, 0xc3, 0x9f, 0x74,
   0x65, 0x6e, 0x2e, 0x22, 0x20, 0x2d, 0x20, 0x22,
   0x47, 0x72, 0xc3, 0xb6, 0xc3, 0x9f, 0x74, 0x65,
   0x6e, 0x20, 0x77, 0x61, 0x73, 0x20, 0x3f, 0x22
};


int   main( int argc, const char * argv[])
{
   NSString       *s;
   NSString       *copy;
   unichar        buf[ 128];
   unsigned int   i;

   for( i = 0; i <= sizeof( hoehoe); i++)
   {
      if( ! [NSString _areValidUTF8Characters:hoehoe
                                       length:i])
      {
         continue;
      }
      s = [NSString _stringWithUTF8Characters:hoehoe
                                       length:i];
      [s getCharacters:buf];
      copy = [NSString stringWithCharacters:buf
                                     length:[s length]];
      printf( "%s (%ld/%ld) -> %s (%ld/%ld)\n",
                  [s UTF8String], [s length], [s _UTF8StringLength],
                  [copy UTF8String], [copy length], [copy _UTF8StringLength]);
   }

   return( 0);
}
