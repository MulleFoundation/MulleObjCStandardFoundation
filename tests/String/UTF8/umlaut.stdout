 (0/0) ->  (0/0)
" (1/1) -> " (1/1)
"H (2/2) -> "H (2/2)
"Hö (3/4) -> "Hö (3/4)
"Höh (4/5) -> "Höh (4/5)
"Höhö (5/7) -> "Höhö (5/7)
"Höhöh (6/8) -> "Höhöh (6/8)
"Höhöhö (7/10) -> "Höhöhö (7/10)
"Höhöhö, (8/11) -> "Höhöhö, (8/11)
"Höhöhö,  (9/12) -> "Höhöhö,  (9/12)
"Höhöhö, w (10/13) -> "Höhöhö, w (10/13)
"Höhöhö, wi (11/14) -> "Höhöhö, wi (11/14)
"Höhöhö, wir (12/15) -> "Höhöhö, wir (12/15)
"Höhöhö, wir  (13/16) -> "Höhöhö, wir  (13/16)
"Höhöhö, wir s (14/17) -> "Höhöhö, wir s (14/17)
"Höhöhö, wir si (15/18) -> "Höhöhö, wir si (15/18)
"Höhöhö, wir sin (16/19) -> "Höhöhö, wir sin (16/19)
"Höhöhö, wir sind (17/20) -> "Höhöhö, wir sind (17/20)
"Höhöhö, wir sind  (18/21) -> "Höhöhö, wir sind  (18/21)
"Höhöhö, wir sind d (19/22) -> "Höhöhö, wir sind d (19/22)
"Höhöhö, wir sind di (20/23) -> "Höhöhö, wir sind di (20/23)
"Höhöhö, wir sind die (21/24) -> "Höhöhö, wir sind die (21/24)
"Höhöhö, wir sind die  (22/25) -> "Höhöhö, wir sind die  (22/25)
"Höhöhö, wir sind die G (23/26) -> "Höhöhö, wir sind die G (23/26)
"Höhöhö, wir sind die Gr (24/27) -> "Höhöhö, wir sind die Gr (24/27)
"Höhöhö, wir sind die Grö (25/29) -> "Höhöhö, wir sind die Grö (25/29)
"Höhöhö, wir sind die Größ (26/31) -> "Höhöhö, wir sind die Größ (26/31)
"Höhöhö, wir sind die Größt (27/32) -> "Höhöhö, wir sind die Größt (27/32)
"Höhöhö, wir sind die Größte (28/33) -> "Höhöhö, wir sind die Größte (28/33)
"Höhöhö, wir sind die Größten (29/34) -> "Höhöhö, wir sind die Größten (29/34)
"Höhöhö, wir sind die Größten. (30/35) -> "Höhöhö, wir sind die Größten. (30/35)
"Höhöhö, wir sind die Größten." (31/36) -> "Höhöhö, wir sind die Größten." (31/36)
"Höhöhö, wir sind die Größten."  (32/37) -> "Höhöhö, wir sind die Größten."  (32/37)
"Höhöhö, wir sind die Größten." - (33/38) -> "Höhöhö, wir sind die Größten." - (33/38)
"Höhöhö, wir sind die Größten." -  (34/39) -> "Höhöhö, wir sind die Größten." -  (34/39)
"Höhöhö, wir sind die Größten." - " (35/40) -> "Höhöhö, wir sind die Größten." - " (35/40)
"Höhöhö, wir sind die Größten." - "G (36/41) -> "Höhöhö, wir sind die Größten." - "G (36/41)
"Höhöhö, wir sind die Größten." - "Gr (37/42) -> "Höhöhö, wir sind die Größten." - "Gr (37/42)
"Höhöhö, wir sind die Größten." - "Grö (38/44) -> "Höhöhö, wir sind die Größten." - "Grö (38/44)
"Höhöhö, wir sind die Größten." - "Größ (39/46) -> "Höhöhö, wir sind die Größten." - "Größ (39/46)
"Höhöhö, wir sind die Größten." - "Größt (40/47) -> "Höhöhö, wir sind die Größten." - "Größt (40/47)
"Höhöhö, wir sind die Größten." - "Größte (41/48) -> "Höhöhö, wir sind die Größten." - "Größte (41/48)
"Höhöhö, wir sind die Größten." - "Größten (42/49) -> "Höhöhö, wir sind die Größten." - "Größten (42/49)
"Höhöhö, wir sind die Größten." - "Größten  (43/50) -> "Höhöhö, wir sind die Größten." - "Größten  (43/50)
"Höhöhö, wir sind die Größten." - "Größten w (44/51) -> "Höhöhö, wir sind die Größten." - "Größten w (44/51)
"Höhöhö, wir sind die Größten." - "Größten wa (45/52) -> "Höhöhö, wir sind die Größten." - "Größten wa (45/52)
"Höhöhö, wir sind die Größten." - "Größten was (46/53) -> "Höhöhö, wir sind die Größten." - "Größten was (46/53)
"Höhöhö, wir sind die Größten." - "Größten was  (47/54) -> "Höhöhö, wir sind die Größten." - "Größten was  (47/54)
"Höhöhö, wir sind die Größten." - "Größten was ? (48/55) -> "Höhöhö, wir sind die Größten." - "Größten was ? (48/55)
"Höhöhö, wir sind die Größten." - "Größten was ?" (49/56) -> "Höhöhö, wir sind die Größten." - "Größten was ?" (49/56)
